src/python/pants/option/global_options.py

```diff
@@ -222,6 +222,13 @@ def register_bootstrap_options(cls, register):
     register('--enable-pantsd', advanced=True, type=bool, default=False,
              help='Enables use of the pants daemon (and implicitly, the v2 engine). (Beta)')
 
+    # Shutdown pantsd after the current run.
+    # This needs to be accessed at the same time as enable_pantsd,
+    # so we register it at bootstrap time.
+    register('--shutdown-pantsd-after-run', advanced=True, type=bool, default=False,
+      help='Shutdown pantsd after the current run.'
```

Comment:
Need a space at the end of this line to avoid concatenating as `run.Will`.

The two sentences also don't seem super consistent... saying "will shutdown after this run" sortof implies that it is already running, but the next sentence says that that isn't legal. Would say something like: "starts, uses, and then stops pantsd for each run".

Also, rather than needing to clarify that this "won't work if pantsd was already running", it might be easier to just implement something whereby the client does something like:
```
if shutdown_pantsd_after_run and pantsd.running():
  pantsd.terminate()
```
I don't think that it would add much latency, assuming that the "running" check is just a PID lookup and check.